exp/etcdrestore/api/v1alpha1/etcdsnapshotrestore_types.go

```diff
@@ -44,41 +45,53 @@ const (
 	ETCDSnapshotRestorePhaseFinished ETCDSnapshotRestorePhase = "Done"
 )
 
-// EtcdSnapshotRestoreSpec defines the desired state of EtcdSnapshotRestore.
-type EtcdSnapshotRestoreSpec struct {
-	ClusterName             string                 `json:"clusterName"`
-	EtcdMachineSnapshotName string                 `json:"etcdMachineSnapshotName"`
-	TTLSecondsAfterFinished int                    `json:"ttlSecondsAfterFinished"`
-	ConfigRef               corev1.ObjectReference `json:"configRef"`
+// +kubebuilder:validation:XValidation:message="Cluster Name can't be empty.",rule="size(self.clusterName)>0"
+// +kubebuilder:validation:XValidation:message="ETCD machine snapshot name can't be empty.",rule="size(self.etcdMachineSnapshotName)>0"
+//
+// ETCDSnapshotRestoreSpec defines the desired state of EtcdSnapshotRestore.
+type ETCDSnapshotRestoreSpec struct {
+	// +required
+	ClusterName string `json:"clusterName"`
+
+	// +required
+	ETCDMachineSnapshotName string `json:"etcdMachineSnapshotName"`
+
+	// TTLSecondsAfterFinished int `json:"ttlSecondsAfterFinished"`
```

Comment:
why are they commented out?

Reply:
We should address this as a part of another issue #764